tests/integration/container_engine.py

```diff
@@ -96,11 +99,140 @@ def run(
+
+        Buildah does not automatically use the image's configured CMD or ENTRYPOINT when running a command.
+        This means that the run command needs to be generated manually considering the image's configuration
+        and the flags present in the original run command.
+
+        :param container_name: The container name to run the command on.
+        :param image: The image on which the command will be run.
+        :param cmd: The original command provided.
+        :param flags: The flags present in the run command.
+        :return: The generated command.
+        """
+        filtered_flags, entrypoint = self._filter_entrypoint_flags(flags)
+        image_cmd, image_entrypoint = self._get_image_config(image)
+
+        # fallback to the image's default cmd if a custom one is not provided
+        cmd = cmd or image_cmd
+
+        # if an entrypoint flag is provided, use it
+        if entrypoint:
+            return ["run", *filtered_flags, container_name, "--", entrypoint, *cmd]
+
+        # if the image has an entrypoint, prepend it to the command
+        if image_entrypoint:
+            return ["run", *filtered_flags, container_name, "--", *image_entrypoint, *cmd]
+
+        return ["run", *filtered_flags, container_name, "--", *cmd]
+
+    def _filter_entrypoint_flags(
+        self, flags: Optional[list[str]]
+    ) -> tuple[list[str], Optional[str]]:
+        """Remove and return any entrypoint flags.
+
+        Since buildah does not support the --entrypoint flag, we need to extract it so that
+        the run command can be properly formatted later on.
+
+        :param flags: The flags to treat.
+        :return: The filtered flags and the entrypoint value.
+        """
+        if flags is None:
+            return [], None
+
+        entrypoint_index = next(
+            (i for i, flag in enumerate(flags) if flag.startswith("--entrypoint")), None
+        )
+
+        if entrypoint_index is None:
+            return flags, None
+
+        entrypoint_flag = flags[entrypoint_index]
+
+        # entrypoint flag is in the format --entrypoint=<entrypoint>
+        if "=" in entrypoint_flag:
+            entrypoint_value = entrypoint_flag.split("=")[1]
+            return flags[:entrypoint_index] + flags[entrypoint_index + 1 :], entrypoint_value
+        # entrypoint flag is in the format --entrypoint <entrypoint>
+        elif entrypoint_index + 1 < len(flags):
+            entrypoint_value = flags[entrypoint_index + 1]
+            return flags[:entrypoint_index] + flags[entrypoint_index + 2 :], entrypoint_value
+        # entrypoint flag exists but no value is provided
+        else:
+            raise RuntimeError(
+                f"Unable to parse the run command flags: {flags}."
+                "Check if the --entrypoint flag defines a value."
+            )
+
+    def _get_image_config(self, image: str) -> tuple[list[str], list[str]]:
+        """Get the configured entrypoint and cmd of an image."""
```

Comment:
nitpick: Replace ` Get the configured entrypoint and cmd of an image.` with `Parse entrypoint and cmd from image's JSON configuration`.